pkg/reconciler/dynamicrestmapper/dynamicrestmapper_builtin_controller.go

```diff
@@ -201,6 +201,9 @@ func (c *BuiltinTypesController) gatherGVKRsForCRD(crd *apiextensionsv1.CustomRe
 }
 
 func (c *BuiltinTypesController) gatherGVKRsForMappedGroupResource(gr schema.GroupResource) ([]typeMeta, error) {
+	c.state.lock.RLock()
```

Comment:
Let's just move the `BuiltinTypesController.gatherGVKRsForMappedGroupResource` call in `BuiltinTypesController.process` after `c.state.lock.Lock()` otherwise there's a race between reading (by `getGVKRs`) and writing (by `apply`).

I see the same issue in `DynamicTypesController` too where write-after-read is not atomic, but this can be fixed in a separate PR.